tests/integration/test_charm.py

```diff
@@ -0,0 +1,27 @@
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = "seldon-controller-manager"
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build and deploy the charm.
+
+    Assert on the unit status.
+    """
+    charm_under_test = await ops_test.build_charm(".")
+    image_path = METADATA["resources"]["oci-image"]["upstream-source"]
+    resources = {"oci-image": image_path}
+
+    await ops_test.model.deploy(
+        charm_under_test, resources=resources, application_name=APP_NAME, trust=True
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60 * 10
+    )
+    assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
```

Comment:
Yes and yes, ensuring the charm can be built and deployed is the only thing this test case is checking. I did it on purpose because I saw that in the previous workflow we are building, deploying and applying the example, but we are not really checking the SeldonDeployment is actually created and is healthy, just that `kubectl apply` is not failing. I think we should have more extensive testing than just building and deploying, but if you see, this is not much different than what we currently have. Adding proper integration testing is a more dedicated task that we should do in the near future outside of this PR.

Reply:
I think it would be good to keep the basic test in place of "can a SeldonDeployment be created", as I'm a little dubious that the upstream labels you linked to [here](https://github.com/kubeflow/manifests/blob/master/contrib/seldon/seldon-core-operator/base/resources.yaml#L4830) work correctly. Even if we don't add any new testing and just do the lightkube equivalent of `kubectl apply`, that should catch any weirdness with the `Service` being set up incorrectly

Reply:
Ah, I see. +1, I'll work on something close to the `kubectl apply` statement we had.